src/core/pointcloud/qgspointcloudblockhandle.h

```diff
@@ -0,0 +1,59 @@
+/***************************************************************************
+                         qgspointcloudblockhandle.h
+                         --------------------
+    begin                : March 2021
+    copyright            : (C) 2021 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPOINTCLOUDBLOCKHANDLE_H
+#define QGSPOINTCLOUDBLOCKHANDLE_H
+
+#include <QObject>
+
+#include "qgspointcloudattribute.h"
+#include "qgstiledownloadmanager.h"
+
+#define SIP_NO_FILE
+
+class QgsPointCloudAttributeCollection;
+class QgsPointCloudBlock;
+
+/**
+ * \ingroup core
+ * \brief Base class for handling loading QgsPointCloudBlock asynchronously
+ *
+ * \note The API is considered EXPERIMENTAL and can be changed without a notice
```

Comment:
or even private header, does it need to be in public api?

Reply:
In case someone wants to use the remote EPT, this needs to be public I think